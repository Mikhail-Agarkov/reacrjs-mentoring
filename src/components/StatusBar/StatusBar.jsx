import React from 'react';
import Filter from "../Filter/Filter";
import styles from './status.module.scss'

export default class StatusBar extends React.PureComponent {

    sortByHandler = (type) => {
        //Api call
    };

    render() {
        console.log(styles.content);
        return (
            <section className={styles.content}>
                <Filter title='Sort by' buttons={
                    [
                        {text: 'Release date', handler: () => this.sortByHandler('Release date'), key: 1},
                        {text: 'Rating', handler: () => this.sortByHandler('Rating'), key: 2}
                    ]
                }/>
            </section>
        );
    }

}