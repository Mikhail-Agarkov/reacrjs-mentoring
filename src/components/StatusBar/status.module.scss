.content {
  height: 70px;
  background-color: #555555;
  overflow: hidden;
}