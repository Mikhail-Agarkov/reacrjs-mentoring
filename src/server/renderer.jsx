import React from 'react';
import {renderToString} from 'react-dom/server';
import {StaticRouter} from 'react-router-dom';
import configureStore from '../redux/store';
import App from "../App";

function renderHTML(html, preLoadedState) {
    return `
      <!doctype html>
      <html lang="en">
        <head>
          <meta charset="UTF-8">
          <meta name="viewport"
            content="width=device-width, user-scalable=no, initial-scale=1.0, maximum-scale=1.0, minimum-scale=1.0">
          <meta http-equiv="X-UA-Compatible" content="ie=edge">
          <link href="https://fonts.googleapis.com/css?family=Montserrat:100,200,300" rel="stylesheet">
          <title>Netflix Roulette</title>
          ${process.env.NODE_ENV === 'development' ? '' : '<link href="/css/main.css" rel="stylesheet" type="text/css">'}
          <script src="/js/main.js" defer></script>
        </head>
        <body>
          <div id="root">${html}</div>
          <script>
            window.PRE_LOADED_STATE = ${JSON.stringify(preLoadedState).replace(/</g, '\\u003c')}
          </script>
        </body>
      </html>
  `;
}

export default function serverRenderer() {
    return (req, res) => {
        const initialState = {
            searchBy: 'title',
            sortBy: 'release_date'
        };
        const store = configureStore(initialState);
        const context = {};

        const renderRoot = () => (
            <App
                context={context}
                location={req.url}
                Router={StaticRouter}
                store={store}
            />
        );

        renderToString(renderRoot());

        if (context.url) {
            res.writeHead(302, {
                Location: context.url,
            });
            res.end();
            return;
        }

        const htmlString = renderToString(renderRoot());
        const preLoadedState = store.getState();

        res.send(renderHTML(htmlString, preLoadedState));
    };
}
