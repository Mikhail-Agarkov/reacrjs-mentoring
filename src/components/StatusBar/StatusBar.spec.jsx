import React from 'react';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';

import StatusBar from './StatusBar';

const mockStore = configureStore([]);

describe('StatusBar', () => {

    let store;
    let component;

    beforeEach(() => {
        store = mockStore({
            sortBy: 'release_date'
        });

        store.dispatch = jest.fn();

        component = renderer.create(
            <Provider store={store}>
                <StatusBar/>
            </Provider>
        );
    });

    it('should render with given state from Redux store', () => {
        expect(component.toJSON()).toMatchSnapshot();
    });

    it('should dispatch an action on button click', () => {
        renderer.act(() => {
            component.root.findAllByType('button')[0].props.onClick();
        });
        expect(store.dispatch).toHaveBeenCalledTimes(1);
    });

    it('should dispatch an action on button click', () => {
        renderer.act(() => {
            component.root.findAllByType('button')[1].props.onClick();
        });
        expect(store.dispatch).toHaveBeenCalledTimes(1);
    });
});