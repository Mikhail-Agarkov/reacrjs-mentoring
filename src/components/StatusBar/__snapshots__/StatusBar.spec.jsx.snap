// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusBar should render with given state from Redux store 1`] = `
<section
  className="content"
>
  <div
    className="content"
  >
    <span
      className="text"
    >
      Sort by
    </span>
    <div
      className="buttonGroup"
    >
      <button
        className="button"
        data-selected={true}
        onClick={[Function]}
      >
        Release date
      </button>
      <button
        className="button"
        data-selected={false}
        onClick={[Function]}
      >
        Rating
      </button>
    </div>
  </div>
</section>
`;
